torch/_dynamo/compiled_autograd.py

```diff
@@ -490,22 +490,49 @@ def set_node_origin(
 in_compiled_autograd_region = False
 
 
-@contextlib.contextmanager
-def enable(compiler_fn):
-    prior = torch._C._dynamo.compiled_autograd.set_autograd_compiler(
-        functools.partial(AutogradCompilerInstance, compiler_fn)
-    )
-    if snapshot_verbose_logging_enabled():
-        torch._C._dynamo.compiled_autograd.set_verbose_logger(cpp_verbose_log_fn)
-    global compiled_autograd_enabled
-    compiled_autograd_enabled = True
-    try:
-        with torch.autograd.set_multithreading_enabled(False):
-            yield
-    finally:
-        if not prior:
-            compiled_autograd_enabled = False
-        torch._C._dynamo.compiled_autograd.set_autograd_compiler(prior)
+class _EnableContext:
+    def __init__(self, compiler_fn):
+        self.compiler_fn = compiler_fn
+        self.warmup_count = 0
```

Comment:
this needs to reuse the same context manager for compiled autograd's torch.compile API to track warmup count: https://github.com/pytorch/pytorch/blame/dbef2b05b4d81e891f7497f92f730a22bebe445d/torch/_dynamo/eval_frame.py#L563